Modify language for reporting signing state (#24932)
hashicorp/terraform: website/docs/plugins/signing.html.md

---
layout: "registry"
page_title: "Plugin Signing"
sidebar_current: "docs-plugins-signing"
description: |-
  Terraform plugin signing trust levels
---

> # Plugin Signing

Reviewer: This document describes plugins in general, but Terraform's tiered signature verification only applies to providers specifically.

Author: Yeah, in this case i was hedging a little on the future, as this is a URL hard coded in to a shipped binary, was making it a bit more future proof. I can add some language though that currently only provider plugins are authenticated.

> There are four levels of signing and programattic authentication for plugins in Terraform:

Reviewer: "Levels of signing" doesn't really capture what's going on here for me. Alternative lede:

> Terraform providers installed from the Registry are cryptographically signed, and the signature is verified at time of installation. There are three types of provider signatures, each with different trust implications:

* **Signed by HashiCorp** - are built, signed, and supported by HashiCorp. Official provider plugins can 

> typically be used without specifying provider source information in your Terraform configuration.

Reviewer: I'm not sure why we're conflating the optional use of provider source attributes with the meaning of the signature. It seems clearer to me to omit this altogether here (and below).

Author: this was just a copy and paste of copy i wrote for the registry, will amend

* **Signed by Trusted Partners** - are built, signed, and supported by a third party. HashiCorp has 
verified the ownership of the private key and we provide a chain of trust to the CLI to verify this
programatically. To use partner providers in your Terraform configuration, you need to specify the
provider source, typically this is the namespace and name to download from the registry.

> * **Self-signed** - are built, signed, and supported by a third party. HashiCorp does not provide a verification or chain of trust for the signing. You will want to obtain and validate fingerprints manually if you want to ensure you are using a binary you can trust.

Reviewer: > You will want to obtain and validate fingerprints manually if you want to ensure you are using a binary you can trust.

It's not clear to me how a user could take action on this recommendation, unless we give some more detail about how to obtain the key fingerprint, and how to usefully validate it. As I understand it that is the subject of some unspecified future work. Until we have something more concrete to offer here, I think it's better to omit this sentence.

Author: Until we offer an ability to authenticate, perhaps we should print a fingerprint for this self-signed category to the output for manual verification?

Reviewer 2: @paultyng we need to limit scope creep, but we will definitely consider that suggestion for a future enhancement.  We don't have concrete plans yet, but a number of ideas to extend the community provider trust process in later releases.

Author: I added in this PR to output KeyID when its third party signed, maybe that will be sufficient?

> * **Unsigned** - Terraform does support fetching and using unsigned binaries, but you should take extreme care when doing so as no programatic authentication is performed on the downloaded binary.

Author: I'm not sure if this is true, just wanted to cover all the states I saw in code.

Reviewer 3: It might, but not from the Registry, which this doc currently lives in the nav hierarchy. Maybe we could clarify that or group them somehow.

Author: I moved the doc, this is top level now under plugins.

Reviewer: We don't support fetching unsigned providers. The only non-local sources follow the Registry protocol, and signatures are mandatory.

Author: I modified it to say this, let me know your thoughts. I wanted to keep it in there, as we will run unsigned, if not fetch them.

---
layout: "registry"
page_title: "Plugin Signing"
sidebar_current: "docs-plugins-signing"
description: |-
  Terraform plugin signing trust levels
---

# Plugin Signing

~> **Note** Currently only provider plugins fetched from a registry are authenticated.

Terraform providers installed from the Registry are cryptographically signed, and the signature is verified at time of installation. There are three types of provider signatures, each with different trust implications:

* **Signed by HashiCorp** - are built, signed, and supported by HashiCorp.
* **Signed by Trusted Partners** - are built, signed, and supported by a third party. HashiCorp has 
verified the ownership of the private key and we provide a chain of trust to the CLI to verify this
programatically.
* **Self-signed** - are built, signed, and supported by a third party. HashiCorp does not provide a

> verification or chain of trust for the signing. You will want to obtain and validate fingerprints manually

Reviewer 2: ```suggestion
verification or chain of trust for the signature. You may obtain and validate fingerprints manually
```
